backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java

```diff
@@ -0,0 +1,666 @@
+package ai.verta.modeldb.cron_jobs;
+
+import static ai.verta.modeldb.authservice.AuthServiceChannel.isBackgroundUtilsCall;
+
+import ai.verta.modeldb.DatasetVisibilityEnum;
+import ai.verta.modeldb.ModelDBConstants;
+import ai.verta.modeldb.ProjectVisibility;
+import ai.verta.modeldb.WorkspaceTypeEnum;
+import ai.verta.modeldb.authservice.AuthService;
+import ai.verta.modeldb.authservice.RoleService;
+import ai.verta.modeldb.collaborator.CollaboratorOrg;
+import ai.verta.modeldb.entities.CommentEntity;
+import ai.verta.modeldb.entities.DatasetEntity;
+import ai.verta.modeldb.entities.DatasetVersionEntity;
+import ai.verta.modeldb.entities.ExperimentEntity;
+import ai.verta.modeldb.entities.ExperimentRunEntity;
+import ai.verta.modeldb.entities.ProjectEntity;
+import ai.verta.modeldb.entities.versioning.BranchEntity;
+import ai.verta.modeldb.entities.versioning.CommitEntity;
+import ai.verta.modeldb.entities.versioning.RepositoryEntity;
+import ai.verta.modeldb.utils.ModelDBHibernateUtil;
+import ai.verta.uac.ModelResourceEnum;
+import ai.verta.uac.UserInfo;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.TimerTask;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+public class DeleteEntitiesCron extends TimerTask {
+  private static final Logger LOGGER = LogManager.getLogger(DeleteEntitiesCron.class);
+  private final AuthService authService;
+  private final RoleService roleService;
+  private final Integer recordUpdateLimit;
+  private static final String DATASET_GLOBAL_SHARING = "_DATASET_GLOBAL_SHARING";
+  private static final String REPOSITORY_GLOBAL_SHARING = "_REPO_GLOBAL_SHARING";
+
+  public DeleteEntitiesCron(
+      AuthService authService, RoleService roleService, Integer recordUpdateLimit) {
+    this.authService = authService;
+    this.roleService = roleService;
+    this.recordUpdateLimit = recordUpdateLimit;
+  }
+
+  /** The action to be performed by this timer task. */
+  @Override
+  public void run() {
+    LOGGER.info("DeleteEntitiesCron wakeup");
+
+    isBackgroundUtilsCall = true;
```

Comment:
so for the execution of this function auth service will act as isBackgroundUtilsCall, for other calls as well?